tests/unit/ai/test_cursor_provider.py

```
@@ -0,0 +1,309 @@
+        ),
+        pytest.raises(AINotAvailableError, match="agent"),
+    ):
+        CursorProvider()
+
+
+def test_cursor_provider_default_model(provider):
+    """Use auto as the default model."""
+    assert_that(provider.model_name).is_equal_to("auto")
+
+
+def test_cursor_provider_custom_model(_mock_agent_on_path):
```

Ruff (0.15.20)

[warning] Fixture `_mock_agent_on_path` without value is injected as parameter, use `@pytest.mark.usefixtures` instead

(PT019)